core/store/src/lib.rs

```diff
@@ -404,6 +407,29 @@ impl StoreUpdate {
         }
         self.storage.write(self.transaction)
     }
+
+    pub fn apply_change_to_flat_state(&mut self, change: &RawStateChangesWithTrieKey) {
+        match &change.trie_key {
+            TrieKey::DelayedReceipt { .. } | TrieKey::DelayedReceiptIndices => return,
+            _ => {}
+        }
+        let key = change.trie_key.to_vec();
+        let last_change = change
```

Comment:
It seems like you are only handling a single change in a vector changes here.  Is that correct?

Reply:
Right. In the `changes` we aggregate all the changes for the key happened in one block (chunk), and we need to apply only the last one.

Reply:
Oh, that is very subtle.